src/03-arrays-tasks.js

```diff
@@ -504,11 +554,19 @@ function distinct(/* arr */) {
  *    "Poland" => ["Lodz"]
  *   }
  */
-function group(/* array, keySelector, valueSelector */) {
-  throw new Error('Not implemented');
+function group(array, keySelector, valueSelector) {
+  const uniqSelectors = Array.from(new Set(array.map((obj) => keySelector(obj))));
+  const mapArr = uniqSelectors.map((selector) => [selector, []]);
+  array.map((obj) => {
+    const selector = keySelector(obj);
+    const selIndex = mapArr.findIndex((item) => item[0] === selector);
+    mapArr[selIndex][1].push(valueSelector(obj));
+    return null;
+  });
+
+  return mapArr;
```

Comment:
Should it be "Map" collection? 
![image](https://user-images.githubusercontent.com/22891941/200372264-1d291fd7-0d49-48e4-9c2e-e4078458b0c3.png)
https://learn.javascript.ru/map-set